redfish-core/lib/fabric_adapters.hpp

```diff
@@ -276,8 +275,13 @@ inline void getFabricAdapterPCIeSlots(
         }
 
         // Check whether PCIeSlot is associated with chassis
-        doGetFabricAdapterPCIeSlots(aResp, fabricAdapterPath, pcieSlotPaths,
-                                    std::move(callback));
+        doGetFabricAdapterPCIeSlots(
```

Comment:
Why change this?

Reply:
It is mainly for the optimization and coding style reasons.

Previously the parameter `callback` of the function `doGetFabricAdapterPCIeSlots` is defined as `call-by-value` rather than as a `r-value` parameter.

As the result, the argument is actually copied to the stack when the function is being called.

Using `r-value` parameters (like the case when the most areas in bmcweb), the compiler can optimize it by moving it to the called function.

```
doGetFabricAdapterPCIeSlots( ..., const dbus::utility::MapperEndPoints&)> callback);
-->
doGetFabricAdapterPCIeSlots( ..., const dbus::utility::MapperEndPoints&)>&& callback);
```

Then, caller needs to conform it.

Without the caller side change, the caller side 'std::move(callback)' won't make any benefit as the callee still requires to copy it to the stack.

Moreover, it may still the compiler gives an error if we just change the callee's side as "&&" like

```
| ../../../../../../../../../ibm-bmcweb/redfish-core/lib/fabric_adapters.hpp:280:46: error: binding reference of type 'std::function<void(const std::__cxx11::basic_string<char>&, const std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >&)>&&' to 'std::remove_reference<const std::function<void(const std::__cxx11::basic_string<char>&, const std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >&)>&>::type' {aka 'const std::function<void(const std::__cxx11::basic_string<char>&, const std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >&)>'} discards qualifiers
|   280 |                                     std::move(callback));
|       |                                     ~~~~~~~~~^~~~~~~~~~
| ../../../../../../../../../ibm-bmcweb/redfish-core/lib/fabric_adapters.hpp:208:9: note:   initializing argument 4 of 'void redfish::doGetFabricAdapterPCIeSlots(const std::shared_ptr<bmcweb::AsyncResp>&, const std::string&, const dbus::utility::MapperEndPoints&, std::function<void(const std::__cxx11::basic_string<char>&, const std::vector<std::__cxx11::basic_string<char>, std::allocator<std::__cxx11::basic_string<char> > >&)>&&)'
|   206 |     std::function<void(const std::string&,
|       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   207 |                        const dbus::utility::MapperEndPoints&)>&&
|       |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   208 |         callback)
|       |         ^~~~~~~~
| ninja: build stopped: subcommand failed.

```

Functionally, they both will be give the same results, but the code with the change will be more optimized and more following bmcweb coding style.